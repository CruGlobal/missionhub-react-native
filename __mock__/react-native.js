import { NativeModules } from 'react-native';

// From https://github.com/facebook/react-native/issues/18279#issuecomment-374177940
// Will be made unnecessary once https://github.com/facebook/react-native/pull/18718 is released. It doesn't seem to be in 0.55.3
NativeModules.BlobModule = {
  ...NativeModules.BlobModule,
  addNetworkingHandler: jest.fn(),
};
