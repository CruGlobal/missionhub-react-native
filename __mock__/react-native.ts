jest.mock(
  'react-native/Libraries/Utilities/NativePlatformConstantsIOS',
  () => ({
    // @ts-ignore
    ...jest.requireActual(
      'react-native/Libraries/Utilities/NativePlatformConstantsIOS',
    ),
    getConstants: () => ({
      forceTouchAvailable: false,
      interfaceIdiom: 'en',
      isTesting: false,
      osVersion: 'ios',
      reactNativeVersion: { major: 60, minor: 1, patch: 0 },
      systemName: 'ios',
    }),
  }),
);
