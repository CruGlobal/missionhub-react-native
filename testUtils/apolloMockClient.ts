import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema, IMocks } from 'graphql-tools';
import { DocumentNode } from 'graphql';
import { buildClientSchema, IntrospectionQuery } from 'graphql/utilities';
import gql from 'fraql';
import { createMockerFromIntrospection } from 'fraql/mock';

import introspectionQuery from '../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks: IMocks = {}) => {
  const schema = buildClientSchema(
    (introspectionQuery as unknown) as IntrospectionQuery,
  );

  addMockFunctionsToSchema({
    schema,
    mocks: { ...globalMocks, ...mocks },
  });

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new SchemaLink({ schema }),
  });
};

export const mockFragment = <TData>(
  fragmentDocument: DocumentNode,
  options?: {
    mocks: IMocks;
  },
): TData => {
  if (!fragmentDocument.loc) {
    throw 'Unable to get original gql query to create FraQL fragment.';
  }

  return createMockerFromIntrospection(introspectionQuery, {
    mocks: globalMocks,
  }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
