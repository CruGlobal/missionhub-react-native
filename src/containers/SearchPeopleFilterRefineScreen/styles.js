
import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  pageContainer: {
    flex: 1,
    backgroundColor: theme.white,
  },
  list: {
    flex: 1,
    backgroundColor: theme.transparent,
  },
});
