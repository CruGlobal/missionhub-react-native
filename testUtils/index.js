import 'react-native';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Need to import for all translation screens to work properly
import '../src/i18n';

Enzyme.configure({ adapter: new Adapter() });

export const createMockStore = (state = {}) => {
  return {
    getState: jest.fn(() => (state)),
    dispatch: jest.fn((response) => Promise.resolve(response)),
    subscribe: jest.fn(),
  };
};

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (data) => {
  expect(renderer.create(data)).toMatchSnapshot();
};

export const renderShallow = (component, store) => {
  let renderedComponent = shallow(
    component,
    { context: { store: store } }
  );

  // If component has translation wrappers, dive deeper
  while (renderedComponent.is('Translate') || renderedComponent.is('I18n')) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (component, store) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};

export const mockFnWithParams = (obj, method, expectedReturn, ...expectedParams) => {
  return obj[method] = jest.fn(
    (...actualParams) => {
      expect(actualParams).toEqual(expectedParams);
      return expectedReturn;
    }
  );
};
