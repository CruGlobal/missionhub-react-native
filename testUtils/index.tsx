import React, { ReactElement, ReactNode } from 'react';
import 'react-native';
import { Provider } from 'react-redux';
import { Provider as ProviderLegacy } from 'react-redux-legacy';
import thunk from 'redux-thunk';
import configureStore, { MockStore } from 'redux-mock-store';
// eslint-disable-next-line import/named
import { NavigationParams, NavigationProvider } from 'react-navigation';
import { ApolloProvider } from '@apollo/react-hooks';
import { ReactTestRendererJSON } from 'react-test-renderer';
import { render, RenderAPI } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';
import Enzyme, { shallow as enzymeShallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { IMocks } from 'graphql-tools';

import { createApolloMockClient } from './apolloMockClient';
import { createNavigationProp } from './navigationHelpers';

Enzyme.configure({ adapter: new Adapter() });

export const createThunkStore = configureStore([thunk]);

interface RenderWithContextParams {
  initialState?: {};
  store?: MockStore;
  navParams?: NavigationParams;
  mocks?: IMocks;
  noWrappers?: boolean;
}

// Inspiration from https://github.com/kentcdodds/react-testing-library/blob/52575005579307bcfbe7fbe4ef4636147c03c6fb/examples/__tests__/react-redux.js#L69-L80
export function renderWithContext(
  component: ReactElement,
  {
    initialState,
    store = createThunkStore(initialState),
    navParams,
    mocks: mocks = {},
    noWrappers = false,
  }: RenderWithContextParams = {},
) {
  const mockApolloClient = createApolloMockClient(mocks);

  const navigation = createNavigationProp(navParams);

  let renderResult: RenderAPI;
  if (noWrappers) {
    renderResult = render(React.cloneElement(component, { navigation }));
  } else {
    // Warning: don't call any functions in here that return new instances on every call. All the props need to stay the same otherwise rerender won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
      <NavigationProvider value={navigation}>
        <ProviderLegacy store={store}>
          <Provider store={store}>
            <ApolloProvider client={mockApolloClient}>
              {children}
            </ApolloProvider>
          </Provider>
        </ProviderLegacy>
      </NavigationProvider>
    );

    renderResult = render(React.cloneElement(component, { navigation }), {
      wrapper,
    });
  }

  let storedSnapshot: ReactTestRendererJSON | null;
  return {
    ...renderResult,
    store,
    snapshot: () => expect(renderResult.toJSON()).toMatchSnapshot(),
    recordSnapshot: () => (storedSnapshot = renderResult.toJSON()),
    diffSnapshot: () => {
      if (!storedSnapshot) {
        throw new Error(
          'You must call recordSnapshot to store an initial snapshot before calling diffSnapshot',
        );
      }
      expect(
        snapshotDiff(storedSnapshot, renderResult.toJSON()),
      ).toMatchSnapshot();
    },
  };
}

// TODO: Remove all legacy rendering functions below

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (component: ReactElement) => {
  const { toJSON } = render(component);
  expect(toJSON()).toMatchSnapshot();
};

export const renderShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  let renderedComponent = enzymeShallow(
    <ProviderLegacy store={store}>{component}</ProviderLegacy>,
  ).dive();

  // If component has translation wrappers, dive deeper
  while (
    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
    ((renderedComponent.type() as any).displayName || '').startsWith(
      'withI18nextTranslation(',
    )
  ) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};
