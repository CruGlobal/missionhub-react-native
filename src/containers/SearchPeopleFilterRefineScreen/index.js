import React, { Component } from 'react';
import { View, FlatList } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import { navigateBack, navigatePush } from '../../actions/navigation';

import Header from '../Header';
import { IconButton } from '../../components/common';
import FilterItem from '../../components/FilterItem';
import styles from './styles';

function setSelected(items = [], id) {
  return items.map((i) => ({
    ...i,
    selected: i.id === id,
    preview: undefined,
  }));
}

@translate('searchFilterRefine')
export class SearchPeopleFilterRefineScreen extends Component {

  constructor(props) {
    super(props);
    const t = props.t;
    let options = [].concat(props.options);
    const hasSelected = !!options.find((o) => o && o.selected);
    if (!options[0] || options[0].id !== 'any') {
      options.unshift({ id: 'any', text: t('any'), selected: !hasSelected });
    }

    this.state = {
      options,
      selectedDrillDownId: '',
    };

    this.handleSelect = this.handleSelect.bind(this);
    this.handleFilterSelect = this.handleFilterSelect.bind(this);
  }

  handleSelect(item) {
    if (item.drilldown) {
      this.setState({ selectedDrillDownId: item.id });
      this.props.dispatch(navigatePush('SearchPeopleFilterRefine', {
        onFilter: this.handleFilterSelect,
        options: item.drilldown,
      }));
    } else {
      const newOptions = setSelected(this.state.options, item.id);
      this.setState({ options: newOptions });
      this.props.onFilter(item);
    }
  }

  handleFilterSelect(item) {
    const option = this.state.options.find((o) => o.id === this.state.selectedDrillDownId);
    const newOptions = this.state.options.map((o) => ({
      ...o,
      selected: o.id === option.id,
      preview: item.text,
    }));
    this.setState({ options: newOptions });
    this.props.onFilter(item);
  }

  render() {
    const { t, title, dispatch } = this.props;
    return (
      <View style={styles.pageContainer}>
        <Header
          left={
            <IconButton
              name="backIcon"
              type="MissionHub"
              onPress={() => dispatch(navigateBack())} />
          }
          title={title || t('title')}
        />
        <FlatList
          style={styles.list}
          data={this.state.options}
          keyExtractor={(i) => i.id}
          initialNumToRender={15}
          renderItem={({ item }) => (
            <FilterItem
              item={item}
              onSelect={this.handleSelect}
              type={item.drilldown ? 'drilldown' : 'single'}
              isSelected={item.selected}
            />
          )}
        />
      </View>
    );
  }
}

SearchPeopleFilterRefineScreen.propTypes = {
  onFilter: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    text: PropTypes.string,
    drilldown: PropTypes.array,
  })).isRequired,
  title: PropTypes.string,
};

const mapStateToProps = (state, { navigation }) => ({
  ...(navigation.state.params || {}),
});

export default connect(mapStateToProps)(SearchPeopleFilterRefineScreen);
