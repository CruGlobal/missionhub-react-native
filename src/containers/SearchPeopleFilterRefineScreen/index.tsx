import React, { Component } from 'react';
import { View, FlatList } from 'react-native';
import { connect } from 'react-redux-legacy';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';

import { navigatePush } from '../../actions/navigation';
import Header from '../../components/Header';
import FilterItem from '../../components/FilterItem';
import { trackSearchFilter } from '../../actions/analytics';
import { buildTrackingObj, keyExtractorId } from '../../utils/common';
import BackButton from '../BackButton';

import styles from './styles';

// @ts-ignore
function setSelected(items = [], id) {
  return items.map(i => ({
    // @ts-ignore
    ...i,
    selected: i.id === id,
    preview: undefined,
  }));
}

// @ts-ignore
@withTranslation('searchFilterRefine')
export class SearchPeopleFilterRefineScreen extends Component {
  // @ts-ignore
  constructor(props) {
    super(props);
    const t = props.t;
    const options = [].concat(props.options);
    // @ts-ignore
    const hasSelected = !!options.find(o => o && o.selected);
    // @ts-ignore
    if (!options[0] || options[0].id !== 'any') {
      // @ts-ignore
      options.unshift({ id: 'any', text: t('any'), selected: !hasSelected });
    }

    this.state = {
      options,
      selectedDrillDownId: '',
    };

    this.handleSelect = this.handleSelect.bind(this);
    this.handleFilterSelect = this.handleFilterSelect.bind(this);
  }

  // @ts-ignore
  handleSelect(item) {
    if (item.drilldown) {
      this.setState({ selectedDrillDownId: item.id });
      // @ts-ignore
      this.props.dispatch(
        navigatePush(SEARCH_REFINE_SCREEN, {
          onFilter: this.handleFilterSelect,
          options: item.drilldown,
          // @ts-ignore
          trackingObj: buildTrackingObj(
            `search : refine : ${item.id}`,
            'search',
            'refine',
            item.id,
          ),
        }),
      );

      // @ts-ignore
      this.props.dispatch(trackSearchFilter(item.id));
    } else {
      // @ts-ignore
      const newOptions = setSelected(this.state.options, item.id);
      this.setState({ options: newOptions });
      // @ts-ignore
      this.props.onFilter(item);
    }
  }

  // @ts-ignore
  handleFilterSelect(item) {
    // @ts-ignore
    const option = this.state.options.find(
      // @ts-ignore
      o => o.id === this.state.selectedDrillDownId,
    );
    // @ts-ignore
    const newOptions = this.state.options.map(o => ({
      ...o,
      selected: o.id === option.id,
      preview: item.text,
    }));
    this.setState({ options: newOptions });
    // @ts-ignore
    this.props.onFilter(item);
  }

  // @ts-ignore
  renderItem = ({ item }) => (
    <FilterItem
      item={item}
      onSelect={this.handleSelect}
      type={item.drilldown ? 'drilldown' : 'single'}
      isSelected={item.selected}
    />
  );

  render() {
    // @ts-ignore
    const { t, title } = this.props;
    return (
      <View style={styles.pageContainer}>
        <Header left={<BackButton />} title={title || t('title')} />
        <FlatList
          style={styles.list}
          // @ts-ignore
          data={this.state.options}
          keyExtractor={keyExtractorId}
          initialNumToRender={15}
          renderItem={this.renderItem}
        />
      </View>
    );
  }
}

// @ts-ignore
SearchPeopleFilterRefineScreen.propTypes = {
  onFilter: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      text: PropTypes.string,
      drilldown: PropTypes.array,
    }),
  ).isRequired,
  title: PropTypes.string,
};

// @ts-ignore
const mapStateToProps = (state, { navigation }) => ({
  ...(navigation.state.params || {}),
});

export default connect(mapStateToProps)(SearchPeopleFilterRefineScreen);
export const SEARCH_REFINE_SCREEN = 'nav/SEARCH_FILTER_REFINE';
