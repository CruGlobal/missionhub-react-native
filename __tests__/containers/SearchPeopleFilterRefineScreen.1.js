import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import { Provider } from 'react-redux';
import { createMockStore } from '../../testUtils/index';
import SearchPeopleFilterRefineScreen from '../../src/containers/SearchPeopleFilterRefineScreen';
import { createMockNavState, testSnapshot } from '../../testUtils';

const store = createMockStore();

jest.mock('react-native-device-info');

it('renders correctly', () => {
  testSnapshot(
    <Provider store={store}>
      <SearchPeopleFilterRefineScreen navigation={createMockNavState({ onFilter: () => {} })} />
    </Provider>
  );
});