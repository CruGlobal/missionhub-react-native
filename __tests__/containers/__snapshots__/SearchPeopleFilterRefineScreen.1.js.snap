// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "#007398",
            "height": 65,
            "paddingTop": 20,
          },
          Object {
            "elevation": 4,
          },
        ],
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingLeft": 5,
          },
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": undefined,
              },
              null,
              Object {},
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "fontSize": 18,
                  },
                  Array [
                    Object {
                      "margin": 10,
                    },
                    Object {},
                  ],
                ],
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flex": 4,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "500",
            },
          ]
        }
      >
        Refine
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingRight": 5,
          },
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#E6E8EC",
              "borderBottomWidth": 1,
              "padding": 13,
            },
            null,
          ],
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "flex": 1,
              "fontSize": 14,
            },
          ]
        }
      >
        Any
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#007398",
                "fontSize": 18,
              },
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#E6E8EC",
              "borderBottomWidth": 1,
              "padding": 13,
            },
            null,
          ],
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "flex": 1,
              "fontSize": 14,
            },
          ]
        }
      >
        Test 1
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#E6E8EC",
              "borderBottomWidth": 1,
              "padding": 13,
            },
            null,
          ],
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "flex": 1,
              "fontSize": 14,
            },
          ]
        }
      >
        Test 10
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders title correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "#007398",
            "height": 65,
            "paddingTop": 20,
          },
          Object {
            "elevation": 4,
          },
        ],
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingLeft": 5,
          },
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": undefined,
              },
              null,
              Object {},
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#ffffff",
                  },
                  Object {
                    "fontSize": 18,
                  },
                  Array [
                    Object {
                      "margin": 10,
                    },
                    Object {},
                  ],
                ],
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flex": 4,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "500",
            },
          ]
        }
      >
        Test
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingRight": 5,
          },
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#E6E8EC",
              "borderBottomWidth": 1,
              "padding": 13,
            },
            null,
          ],
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "flex": 1,
              "fontSize": 14,
            },
          ]
        }
      >
        Any
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#007398",
                "fontSize": 18,
              },
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#E6E8EC",
              "borderBottomWidth": 1,
              "padding": 13,
            },
            null,
          ],
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "flex": 1,
              "fontSize": 14,
            },
          ]
        }
      >
        Test 1
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#E6E8EC",
              "borderBottomWidth": 1,
              "padding": 13,
            },
            null,
          ],
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "flex": 1,
              "fontSize": 14,
            },
          ]
        }
      >
        Test 10
      </Text>
    </View>
  </View>
</View>
`;
